docs/Subnet-Utilities.rst

```diff
@@ -0,0 +1,110 @@
+Subnet Utilities
+================
+
+ ``Arcus.Utilities.SubnetUtilities`` is a static utility class containing miscellaneous operations for :ref:`Subnet` and collections there of. It is a catchall for methods and functionally that didn't make sense on the ``Subnet`` class itself.
+
+find Fewest Consecutive Subnets
+-------------------------------
+
+Given an inclusive range of IP Addresses defined by ``IPAddress left`` and ``IPAddress right`` get the fewest consecutive subnets that would contain all addresses within the range between and no other addresses.
+
+.. code-block:: c#
+
+   public static IEnumerable<Subnet> FewestConsecutiveSubnetsFor(IPAddress left, IPAddress right)
+
+The following examples shows that the range defined by ``192.168.1.3`` - ``192.168.1.5`` fits in  two consecutive subnets defined by ``192.168.1.4/31`` and ``192.168.1.3/32``.
+
+.. code-block:: c#
+   :emphasize-lines: 9
+   :caption: FewestConsecutiveSubnetsFor Example
+   :name: FewestConsecutiveSubnetsFor Example
+
+    [Fact]
+    public void FewestConsecutiveSubnetsFor_Example()
+    {
+        // Arrange
+        var left = IPAddress.Parse("192.168.1.3");
+        var right = IPAddress.Parse("192.168.1.5");
+
+        // Act
+        var result = SubnetUtilities.FewestConsecutiveSubnetsFor(left, right);
+
+        // Assert
+        Assert.Equal(2, result.Length);
+        Assert.Contains(Subnet.Parse("192.168.1.4/31"), result);
+        Assert.Contains(Subnet.Parse("192.168.1.3/32"), result);
+    }
+
+
+find the Largest Subnet in an enumerable
+----------------------------------------
+
+Given an ``IEnumerable<Subnet>`` find the largest subnet (smallest route prefix value) within.
+
+.. note:: In the case of no single largest subnet one of will be returned. In this case Consider usage of the ``DefaultSubnetComparer``.
```

Comment:
```rst
The ``LargestSubnet` method, given an ``IEnumerable<Subnet>`` will find select first largest subnet from within the collection.

.. note:: If there is no single largest in the input, the first largest subnet encountered will be returned. In cases such as this it may be preferable to consider usage of the ``DefaultSubnetComparer``.
```